# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserHost.cpp
# Commit: 1663719e2aa5
# Full Hash: 1663719e2aa5b6f77e32398c6d6fcb4cb3c999ed
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 16 - Cycle collect RemoteBrowser and support weak references in BrowserHost. r=nika
#   
#   BrowserParent is cycle collected and supported weak references, so this commit adds support
#   for these things to BrowserHost.
#   
# ==============================================================================

diff -r 53663cad8890 -r 1663719e2aa5 dom/ipc/BrowserHost.cpp
--- a/dom/ipc/BrowserHost.cpp	Wed May 08 16:12:26 2019 -0500
+++ b/dom/ipc/BrowserHost.cpp	Tue May 14 16:28:10 2019 -0500
@@ -12,7 +12,16 @@
 namespace mozilla {
 namespace dom {
 
-NS_IMPL_ISUPPORTS(BrowserHost, nsIRemoteTab)
+NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(BrowserHost)
+  NS_INTERFACE_MAP_ENTRY(nsIRemoteTab)
+  NS_INTERFACE_MAP_ENTRY(nsISupportsWeakReference)
+  NS_INTERFACE_MAP_ENTRY_AMBIGUOUS(nsISupports, RemoteBrowser)
+NS_INTERFACE_MAP_END
+
+NS_IMPL_CYCLE_COLLECTION(BrowserHost, mRoot)
+
+NS_IMPL_CYCLE_COLLECTING_ADDREF(BrowserHost)
+NS_IMPL_CYCLE_COLLECTING_RELEASE(BrowserHost)
 
 BrowserHost::BrowserHost(BrowserParent* aParent) : mRoot(aParent) {
   mRoot->SetBrowserHost(this);